packages/flutter_test/lib/src/matchers.dart

```diff
@@ -2502,7 +2510,16 @@ class _MatchesSemanticsData extends Matcher {
       return failWithDescription(matchState, 'No SemanticsData provided. '
         'Maybe you forgot to enable semantics?');
     }
-    final SemanticsData data = node is SemanticsNode ? node.getSemanticsData() : (node as SemanticsData);
+
+    final SemanticsData data;
+    if (node is SemanticsNode) {
+      data = node.getSemanticsData();
+    } else if (node is FinderBase<SemanticsNode>) {
+      data = node.evaluate().single.getSemanticsData();
+    } else {
+      data = node as SemanticsData;
+    }
```

Comment:
optional: This could be rewritten with one of them fancy new switch expressions.

Reply:
Done! I like the fancy switch expressions, but I still have to look up the syntax every time...